File: Leetcode/67. Add Binary.ts
function addBinary(a: string, b: string): string {
    let s: string = "";
    
    a = a.split("").reverse().join("");
    b = b.split("").reverse().join("");
    
    if(a.length > b.length) [a,b] = [b,a];
    
    let c: number = 0;
    
    for(let i: number = 0; i<a.length; i++) {
        let x: number = parseInt(a[i]);
        let y: number = parseInt(b[i]);
        s += ((x+y+c)%2).toString();
        c = Math.floor((x+y+c)/2);
    }
    
    for(let i: number = a.length; i<b.length; i++) {
        let y: number = parseInt(b[i]);
        s += ((y+c)%2).toString();
        c = Math.floor((y+c)/2);
    }
    
    if(c > 0) s += c.toString();
    
    return s.split("").reverse().join("");
};

File: Leetcode/67. Add Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int c = 0, i=0, j=0;
        string ans = "";
        reverse(a.begin(),a.end());
        reverse(b.begin(),b.end());
        
        while(i < a.size() || j < b.size()) {
            if(i >= a.size()) {
                int x = (b[j++] - '0') + c;
                ans += (x%2 + '0');
                c = x/2;
            } else if(j >= b.size()) {
                int x = (a[i++] - '0') + c;
                ans += (x%2 + '0');
                c = x/2;
            } else {
                int x = (a[i++] - '0') + (b[j++] - '0') + c;
                ans += (x%2 + '0');
                c = x/2;
            }
        }
        
        if(c) ans += (c + '0');
        
        reverse(ans.begin(),ans.end());
        
        return ans;
    }
};
